### PR TITLE
feat(cache): dynamodb implementation

### PR DESCRIPTION
**Pull request by chofnar**
**What type of PR is this?**

enhancement

**Which issue does this PR fix**:

Part of #564 

**What does this PR do / Why do we need it**:

Added AWS DynamoDB driver.

**Will this break upgrades or downgrades?**

If a client uses DynamoDB and decides to downgrade, they must make sure that the database is migrated to BoltDB to avoid data loss.

New validations that might break stuff:
local storage driver, but explicit RemoteCache=true, this is not valid, so error out
remote storage driver, but no "cacheDriver" config section, error out

Defaults:
local storage driver -> RemoteCache=false
remote storage driver (s3) -> RemoteCache=true

**Does this PR introduce any user-facing change?**:

Adds the possibility to use AWS DynamoDB instead of BoltDB

**Problems**
- need to specify protocol in front of dynamodb endpoint [(known issue)](https://github.com/aws/aws-sdk-go-v2/issues/1294)

**To do**
- [x] fix lint problems
- [x] remove comment from controller.go line 461
- [x] fix misleading example config (local storage with remote cache, is invalid)
- [x] check that the example config actually works
- [x] add negative tests
- [x] previous tests pass
- [x] add coverage tests

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by peusebiu** [comment]
let's add a new testCase in TestS3Dedupe https://github.com/project-zot/zot/blob/main/pkg/storage/s3/s3_test.go#L935

Currently TestS3Dedupe it's using boltdb, let's use both boltdb and DynamoDB, this should be a small change.

**Comment by rchincha** [comment]
Thanks for tracking your PR with task items, makes it easier to follow.

File: examples/config-dynamodb.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+    "distSpecVersion": "1.0.1-dev",
+    "storage": {
```
Comment (rchincha) [inline examples/config-dynamodb.json:3]: trivy's CVE db is the only one that should take up local disk. If that is disabled in config, can we make sure that nothing is present on local disk except config?

File: .github/workflows/ci-cd.yml [hunk 2]
```diff
@@ -38,7 +38,7 @@ jobs:
       s3mock:
         image: ghcr.io/project-zot/localstack/localstack:1.2.0
         env:
-          SERVICES: s3
+          SERVICES: s3,dynamodb
```
Comment (rchincha) [inline .github/workflows/ci-cd.yml:41]: `localstack` is used to mock s3. What is being used for dynamodb?

Comment (chofnar) [reply .github/workflows/ci-cd.yml:41]: It is used to mock dynamodb for tests

File: examples/config-dynamodb.json (hunk 1 above)
Comment (chofnar) [reply examples/config-dynamodb.json:3]: Do you mean that nothing should be written on the disk by zot if all the config is cloud-based, and a mechanism to check for that should be implemented? I'm not sure I understand

**Pull request merged**